brainiak/reconstruct/iem.py

```diff
@@ -0,0 +1,466 @@
+    channels_: [n_channels, channel density] NumPy 2D array
+        matrix defining channel values
+
+    W_: sklearn.linear_model model containing weight matrix that
+        relates estimated channel responses to response amplitude
+        data
+    """
+    def __init__(self, n_channels=6, channel_exp=5,
+                 stimulus_mode='halfcircular', range_start=0.,
+                 range_stop=180., channel_density=180,
+                 stimulus_resolution=None):
+        self.n_channels = n_channels
+        self.channel_exp = channel_exp
+        self.stimulus_mode = stimulus_mode
+        self.range_start = range_start
+        self.range_stop = range_stop
+        self.channel_density = channel_density
+        self.channel_domain = np.linspace(range_start, range_stop-1,
+                                          channel_density)
+        if stimulus_resolution is None:
+            self.stim_res = channel_density
+        else:
+            self.stim_res = stimulus_resolution
+        self._check_params()
+
+    def _check_params(self):
+        if self.range_start >= self.range_stop:
+            raise ValueError("range_start {} must be less than "
+                             "{} range_stop.".format(self.range_start,
+                                                     self.range_stop))
+        if self.stimulus_mode == 'halfcircular':
+            if (self.range_stop - self.range_start) != 180.:
+                raise ValueError("For half-circular feature spaces,"
+                                 "the range must be 180 degrees, "
+                                 "not {}".format(self.range_stop
+                                                 - self.range_start))
+        elif self.stimulus_mode == 'circular':
+            if (self.range_stop - self.range_start) != 360.:
+                raise ValueError("For circular feature spaces, the"
+                                 " range must be 360 degrees"
+                                 "not {}".format(self.range_stop
+                                                 - self.range_start))
+        if self.n_channels < 2:
+            raise ValueError("Insufficient number of channels.")
+        if not np.isin(self.stimulus_mode, ['circular',
+                                            'halfcircular']):
+            raise ValueError("Stimulus mode must be one of these: "
+                             "'circular', 'halfcircular'")
+
+    def fit(self, X, y):
+        """Use data and feature variable labels to fit an IEM
+
+        Parameters
+        ----------
+        X: numpy matrix of voxel activation data. [observations, voxels]
+            Should contain the beta values for each observation or
+            trial and each voxel of training data.
+        y: numpy array of response variable. [observations]
+            Should contain the feature for each observation in X.
+        """
+        # Check that data matrix is well conditioned:
+        if np.linalg.cond(X) > 9000:
```

Comment:
it might be better to define a variable called `MAX_COND_NUMBER`, since this is re-used later.